### DISCUSSION TITLE
Custom checkout flow example

### DISCUSSION THREAD

**Discussion by vishalshah6153** [swellstores/community#69] [Help] [answered]
Hi, I've been trying to implement the exact same checkout flow that swell provides by default, but I'm stuck with the billing stage. With tokenize method the swell updates cart.billing with card details, but I also need to update the billing address obviously. If I try to create card tokens manually I'm not sure how I can get the data that the user enters into the card element from Stripe. So I'll be very grateful if anyone that has implemented the custom flow already gives me some ideas or suggestions on how I can resolve this issue.

**Comment by y3owk1n** [comment, marked as answer]
@vishalshah6153 I had implemented my own checkout flow as well. If you're referring to `Direct Credit Card Tokenization`, maybe [this](https://developers.swell.is/frontend-api/payments#add-to-cart) would help.

TLDR:
1. You have a form that takes in credit card and billing address
2. You then call the `createToken` fuction by passing along your addresses
3. Attach the `token` response to the cart
4. Swell should do the rest for you
